tests/sentry/utils/suspect_resolutions/test_metric_correlation.py

```diff
@@ -0,0 +1,230 @@
+import random
+from unittest import mock
+
+from django.utils import timezone
+
+from sentry.models import GroupStatus
+from sentry.testutils import TestCase
+from sentry.utils.suspect_resolutions.metric_correlation import (
+    calculate_pearson_correlation_coefficient,
+    is_issue_error_rate_correlated,
+)
+
+START = 1656393600
+END = 1656396780
+WINDOW = 60
+
+
+def generate_issue_error_series(
+    start_timestamp, end_timestamp, data_fill, empty_fill=0, time_window=60
+):
+    return data_fill + [(t, empty_fill) for t in range(start_timestamp, end_timestamp, time_window)]
+
+
+class TestMetricCorrelation(TestCase):
+    @mock.patch("sentry.tsdb.get_range")
+    def test_correlated_issues(self, mock_get_range):
+        group1 = self.create_group(status=GroupStatus.RESOLVED, resolved_at=timezone.now())
+        group2 = self.create_group()
+
+        gen_random_start = 1656393120
+        gen_random_end = 1656393600
+
+        group1_data = [
+            (t, random.randint(0, 30)) for t in range(gen_random_start, gen_random_end, WINDOW)
+        ]
+        group2_data = [
+            (t, random.randint(0, 30)) for t in range(gen_random_start, gen_random_end, WINDOW)
+        ]
```

Comment:
Can we wrap this in a function as well? At a glance, I can't quickly understand what this is doing - assigning a function with a name would help convey some information without adding too many comments.